### PR TITLE
P0920R2 Precalculated hash values in lookup

### LINKED ISSUE #2709: [2019-02 LWG Motion 18] P0920R2 Precalculated hash values in lookup
[P0920R2](https://wg21.link/p0920r2)

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #2709.

### CODE REVIEW

File: source/containers.tex [hunk 1]
```diff
@@ -2794,6 +2843,17 @@
...
 \\ \rowsep
 %
+\tcode{b.equal_range(k, hk)}
+&   \tcode{pair<iterator, iterator>}; \br
+    \tcode{pair<const_iterator, const_iterator>} for const \tcode{b}.
+&   \expects \tcode{b.hash_function()(k)} equals \tcode{hk}.\br
+    \returns a range containing all elements with keys equivalent to
+    \tcode{k}.  Returns \tcode{make_pair(b.end(), b.end())} if
+    no such elements exist.%
+&   Average case \bigoh{\tcode{b.count(k)}}.  Worst case
```
Comment (burblebee) [inline source/containers.tex]: We're inconsitent as to wether we do 
`Average case x.  Worst case y.`
vs
`Average case x, worst case y.`
Should we open an editorial issue for this?

File: source/containers.tex [hunk 2]
```diff
@@ -2293,6 +2293,7 @@
 \item \tcode{n} denotes a value of type \tcode{size_type},
 \item \tcode{z} denotes a value of type \tcode{float}, and
 \item \tcode{nh} denotes a non-const rvalue of type \tcode{X::node_type}.
+\item \tcode{hk} and \tcode{hke} denote values of type \tcode{size_t}.
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/containers.tex]: Fixed.

File: source/containers.tex (hunk 1 above)
Comment (jensmaurer) [reply source/containers.tex]: I would even vouch for a semicolon here. Yes, please make that a separate editorial issue and put "decision-required" on it.

Comment (burblebee) [reply source/containers.tex]: Opened #2749.

**Pull request merged**